webpack/webpack-env.d.ts

```diff
@@ -101,3 +102,137 @@ declare var __non_webpack_require__: any;
  * Equals the config option debug
  */
 declare var DEBUG: boolean;
+
+/**
+ * Webpack Hot Module Replacement
+ * https://webpack.github.io/docs/hot-module-replacement.html
+ */
+declare namespace __WebpackHotModuleReplacement {
+  interface Module {
+    exports: any;
+    require(id: string): any;
+    id: string;
+    filename: string;
+    loaded: boolean;
+    parent: any;
+    children: any[];
+    hot: Hot;
+  }
+
+  interface Hot {
+    /**
+     * Accept code updates for the specified dependencies. The callback is called when dependencies were replaced.
+     * @param dependencies
+     * @param callback
+     */
+    accept(dependencies: string[], callback: (updatedDependencies: string[]) => void): void;
+    /**
+     * Accept code updates for the specified dependencies. The callback is called when dependencies were replaced.
+     * @param dependency
+     * @param callback
+     */
+    accept(dependency: string, callback: () => void): void;
+    /**
+     * Accept code updates for this module without notification of parents.
+     * This should only be used if the module doesn’t export anything.
+     * The errHandler can be used to handle errors that occur while loading the updated module.
+     * @param errHandler
+     */
+    accept(errHandler?: Function): void;
+    /**
+     * Do not accept updates for the specified dependencies. If any dependencies is updated, the code update fails with code "decline".
+     */
+    decline(dependencies: string[]): void;
+    /**
+     * Do not accept updates for the specified dependencies. If any dependencies is updated, the code update fails with code "decline".
+     */
+    decline(dependency: string): void;
+    /**
+     * Flag the current module as not update-able. If updated the update code would fail with code "decline".
+     */
+    decline(): void;
+    /**
+     * Add a one time handler, which is executed when the current module code is replaced.
+     * Here you should destroy/remove any persistent resource you have claimed/created.
+     * If you want to transfer state to the new module, add it to data object.
+     * The data will be available at module.hot.data on the new module.
+     * @param callback
+     */
+    dispose(callback: (data: any) => void): void;
```

Comment:
Should generally prefer type parameters instead of `any`:
```typescript
dispose<T>(callback:(data: T) => void): void;
```

This will default `T` to `{}` and ensure type safety.